controllers/util/provider.go

```diff
@@ -110,6 +122,18 @@ func GetProviderCredentials(ctx context.Context, k8sClient client.Client, namesp
 				EnvAWSSecretAccessKey: ak.AWSSecretAccessKey,
 				EnvAWSDefaultRegion:   region,
 			}, nil
+		case string(GCP):
+        	var ak GCPCredentials
+        	if err := yaml.Unmarshal(secret.Data[secretRef.Key], &ak); err != nil {
+            	errMsg := "failed to convert the credentials of Secret from Provider"
+        		klog.ErrorS(err, errMsg, "Name", secretRef.Name, "Namespace", secretRef.Namespace)
+        		return nil, errors.Wrap(err, errMsg)
+        	}
+        	return map[string]string{
+        	    EnvGCPCredentialsJSON: ak.GCPCredentialsJSON,
```

Comment:
`EnvGCPCredentialsJSON` is a JSON file, how is it imported to the container of terraform controller?

Reply:
You just need to export an Environment Variable with JSON inside.
Example:

`export GOOGLE_CREDENTIALS='{                                                                                                                                                                                                      
  "type": "service_account",
  "project_id": "example-project-123456", 
  "private_key_id": "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
  "private_key": "-----BEGIN PRIVATE KEY-----\\n-----END PRIVATE KEY-----\n",
  "client_email": "test@developer.gserviceaccount.com",
  "client_id": "123456789012345678901",
  "auth_uri": "https://accounts.google.com/o/oauth2/auth",
  "token_uri": "https://oauth2.googleapis.com/token",
  "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
  "client_x509_cert_url": "https://www.googleapis.com/robot/v1/metadata/x509/test%40developer.gserviceaccount.com"
}'
`

Reply:
Got it. I've checked Terraform site https://registry.terraform.io/providers/hashicorp/google/latest/docs/guides/provider_reference. It seems GCP credential is not friendly to Terraform, compared to Aws/Azure/Alibaba Cloud. Let me continue the verification. Thanks.

Reply:
Exact. Right now the fastest way is to use the environment variable with the JSON inside.
Later, it may be useful to mount the credentials file on the controller. What do you think about it?

Reply:
+1.

Reply:
> You just need to export an Environment Variable with JSON inside.
> Example:
> 
> `export GOOGLE_CREDENTIALS='{ "type": "service_account", "project_id": "example-project-123456", "private_key_id": "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa", "private_key": "-----BEGIN PRIVATE KEY-----\\n-----END PRIVATE KEY-----\n", "client_email": "test@developer.gserviceaccount.com", "client_id": "123456789012345678901", "auth_uri": "https://accounts.google.com/o/oauth2/auth", "token_uri": "https://oauth2.googleapis.com/token", "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs", "client_x509_cert_url": "https://www.googleapis.com/robot/v1/metadata/x509/test%40developer.gserviceaccount.com" }' `

Can you give an elaborated example when exporting `GOOGLE_CREDENTIALS` in `getting-started.md` by removing some sensitive information from the value of key `GOOGLE_CREDENTIALS`?

Reply:
Done!